1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/ItemsDataGenerator.java

```diff
@@ -45,11 +45,12 @@ public static JsonObject generateItem(Registry<Item> itemRegistry, Item item) {
         itemDesc.addProperty("displayName", DGU.translateText(item.getTranslationKey()));
         itemDesc.addProperty("stackSize", item.getMaxCount());
 
-        List<TagKey<Item>> enchantmentTargets = getApplicableEnchantmentTargets(item);
+        List<RegistryEntryList<Item>> enchantmentTargets = getApplicableEnchantmentTargets(item);
 
         JsonArray enchantCategoriesArray = new JsonArray();
-        for (TagKey<Item> target : enchantmentTargets) {
-            enchantCategoriesArray.add(EnchantmentsDataGenerator.getEnchantmentTargetName(target));
+        for (RegistryEntryList<Item> target : enchantmentTargets) {
+            String category = EnchantmentsDataGenerator.getEnchantmentTargetName(target);
+            enchantCategoriesArray.add(category);
```

Comment:
This doesn't work, enchantCategories is always empty.
I have done this code that works but may not be the cleanest, I'm not used to fabric development.
```java
        JsonArray enchantCategoriesArray = new JsonArray();
        Registry<Enchantment> enchants = DGU.getWorld().getRegistryManager().get(RegistryKeys.ENCHANTMENT);
        for (Enchantment enchant : enchants) {
            if (enchant.getApplicableItems().contains(item.getRegistryEntry())) {
                String enchantTarget = enchant.getApplicableItems().getTagKey().get().id().getPath().split("/")[1];
                if (!enchantCategoriesArray.contains(new JsonPrimitive(enchantTarget))) {
                    enchantCategoriesArray.add(enchantTarget);
                }
            }
        }
```